test/validation/api/crypto/odp_crypto_test_inp.c

```diff
@@ -1095,12 +1132,187 @@ static void test_capability(void)
 	CU_ASSERT((~capa.auths.all_bits & capa.hw_auths.all_bits) == 0);
 }
 
+/*
+ * Create a test reference, which can be used in tests where the hash
+ * result is within auth_range.
+ *
+ * The ciphertext packet and the hash are calculated using an encode
+ * operation with hash_result_offset outside the auth_range and by
+ * copying the hash in the ciphertext packet.
+ */
+static void create_hash_test_reference(odp_auth_alg_t auth,
+				       const odp_crypto_auth_capability_t *capa,
+				       crypto_test_reference_t *ref,
+				       uint32_t digest_offset,
+				       uint8_t digest_fill_byte)
+{
+	odp_crypto_session_t session;
+	uint32_t n;
+	int rc;
+	odp_packet_t pkt;
+	odp_bool_t ok;
+	const uint32_t auth_bytes = 100;
+	uint32_t enc_digest_offset = auth_bytes;
+	odp_packet_data_range_t cipher_range = {.offset = 0, .length = 0};
+	odp_packet_data_range_t auth_range = {.offset = 0};
+
+	ref->auth_key_length = capa->key_len;
+	ref->auth_iv_length = capa->iv_len;
+	ref->digest_length = capa->digest_len;
+	if (capa->bit_mode)
+		ref->length = auth_bytes * 8;
+	else
+		ref->length = auth_bytes;
+	auth_range.length = ref->length;
+
+	for (n = 0; n < ref->auth_key_length; n++)
+		ref->auth_key[n] = n;
+	for (n = 0; n < ref->auth_iv_length; n++)
+		ref->auth_iv[n] = n;
+	for (n = 0; n < ref->length; n++)
```

Comment:
A fill-like helper function could be useful for these.

Reply:
Will add in v2.